src/WinGet.RestSource.IntegrationTest/Tests/Winget/WingetTests.cs

```diff
@@ -0,0 +1,156 @@
+        public async Task InitializeAsync()
+        {
+            if (this.addRestSource)
+            {
+                // Must be running as admin to succeed.
+                await this.AddSource();
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task DisposeAsync()
+        {
+            if (this.addRestSource)
+            {
+                // Must be running as admin to succeed.
+                await this.RemoveSource();
+            }
+        }
+
+        /// <summary>
+        /// Tests winget's upgrade command against the REST api.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task WingetUpgrade()
+        {
+            await this.TestWingetQuery("upgrade Microsoft.PowerShell", output => output.Contains("No applicable update found."));
+            await this.TestWingetQuery("upgrade Microsoft.PowerShelllll", output => output.Contains("No installed package found matching input criteria."));
+        }
+
+        /// <summary>
+        /// Tests winget's list command against the REST api.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task WingetList()
+        {
+            await this.TestWingetQuery("list \"Windows Software Development Kit\"", output => output.Contains("Microsoft.WindowsSDK"));
+        }
+
+        /// <summary>
+        /// Tests winget's search command against the REST api.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task WingetSearch()
+        {
+            await this.TestWingetSearchQuery("PowerToys", PowerToysPackageIdentifier);
+            await this.TestWingetSearchQuery("powertoys", PowerToysPackageIdentifier);
+            await this.TestWingetSearchQuery("PowerT", PowerToysPackageIdentifier);
+            await this.TestWingetSearchQuery("owertoy", PowerToysPackageIdentifier);
+            await this.TestWingetSearchQuery("nonexistentpackage");
+
+            await this.TestWingetSearchQuery("--name PowerToys", PowerToysPackageIdentifier);
+            await this.TestWingetSearchQuery("--name powertoys", PowerToysPackageIdentifier);
+            await this.TestWingetSearchQuery("--name PowerT", PowerToysPackageIdentifier);
+            await this.TestWingetSearchQuery("--name owertoy", PowerToysPackageIdentifier);
+            await this.TestWingetSearchQuery("--name nonexistentpackage");
+        }
+
+        private static IEnumerable<WingetApp> ParseWingetOutput(string output)
+        {
+            var rows = output.Split(Environment.NewLine);
+
+            if (rows[0].Contains("No package found matching input criteria."))
+            {
+                return Enumerable.Empty<WingetApp>();
+            }
+
+            return rows
```

Comment:
Might be nice to leave a comment showing example output this expects to parse.  Oneday, this will break :)

Reply:
Good point